src/common/reducers/snaps.js

```diff
@@ -31,6 +58,11 @@ export function snaps(state = {
         success: false,
         error: action.payload
       };
+    case RegisterNameActionTypes.REGISTER_NAME_SUCCESS:
+      return {
+        ...state,
+        snaps: updateRegisteredName(state.snaps, action.payload.id, action.payload.snapName)
```

Comment:
Non-blocking comment: We should look at replacing stores with a proper data structure lib. It would prevent us having to write functions to traverse and update data structures everywhere it is needed.